dcm4che-core/src/main/java/org/dcm4che3/io/stream/ImageInputStreamLoader.java

```diff
@@ -0,0 +1,27 @@
+//
+///////////////////////////////////////////////////////////////
+//                C O P Y R I G H T  (c) 2019                //
+//        Agfa HealthCare N.V. and/or its affiliates         //
+//                    All Rights Reserved                    //
+///////////////////////////////////////////////////////////////
+//                                                           //
+//       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF      //
+//        Agfa HealthCare N.V. and/or its affiliates.        //
+//      The copyright notice above does not evidence any     //
+//     actual or intended publication of such source code.   //
+//                                                           //
+///////////////////////////////////////////////////////////////
+//
+package org.dcm4che3.io.stream;
+
+import javax.imageio.stream.ImageInputStream;
+import java.io.IOException;
+
+
+public interface ImageInputStreamLoader<T> {
```

Comment:
Interface to wrap the creation of an ImageInputStream from an input value.  This is required to support delegating of IIS input, but it also allows us to hide the details of service loader binding which can get a bit ugly.